src/System.IO.FileSystem/src/System/IO/PathHelpers.Windows.cs

```diff
@@ -10,7 +10,7 @@ internal static partial class PathHelpers
     {
         // Trim trailing whitespace, tabs etc but don't be aggressive in removing everything that has UnicodeCategory of trailing space.
         // String.WhitespaceChars will trim more aggressively than what the underlying FS does (for ex, NTFS, FAT).    
-        internal static readonly char[] TrimEndChars = { (char)0x9, (char)0xA, (char)0xB, (char)0xC, (char)0xD, (char)0x20, (char)0x85, (char)0xA0 };
+        internal static readonly char[] TrimEndChars = { (char)0x9, (char)0xA, (char)0xB, (char)0xC, (char)0xD, (char)0x20 };
```

Comment:
Kill the whole thing- it isn't correct or useful. Being "helpful" like this causes blocks like yours. There are way too many obscure paths that can come into play that don't play by NTFS or FAT rules (pipes, mailslots, etc.).  :)